src/popup/fetch.ts

```diff
@@ -26,6 +27,22 @@ function roundToNextHour(date: Date): Date {
 	return date;
 }
 
+function reformatDate(dateString: string, timeZone: string | null): string {
+	/*
+		Problem:  Notion does not convert times into the correct timezone,
+				  even when supplied a timezone, Notion will show the user the
+				  UTC time (incorrect time, unless you live in UTC region), labeled
+				  with the supplied timezone.
+		Solution: Re-format the ISO date string to give Notion a 'false' date string based
+				  on the timezone provided by the user. This is a misuse of ISO time/date strings,
+				  but it will result in the correct date being shown in Notion.
+	 */
+	const date = new Date(dateString);
+	const offset = moment.tz.zone(timeZone ?? 'Pacific/Auckland')?.utcOffset(date.valueOf()) ?? 0;
```

Comment:
This looks really good, thank you! Out of curiosity---are you able to get this to default to the user's timezone instead?

I think `timeZone ?? moment.tz.guess()` should work ([docs](https://momentjs.com/timezone/docs/#/using-timezones/guessing-user-timezone/)).